core/src/main/clojure/xtdb/indexer.clj

```diff
@@ -455,9 +455,34 @@
                                  (map-indexed (fn [idx ^IVectorReader col]
                                                 (.withName col (str "?_" idx))))))))))))
 
+(def ^:private ^:const ^String user-table "pg_catalog/pg_user")
+
+(defn- update-pg-user! [^ILiveIndexTx live-idx-tx, ^TransactionKey tx-key, user, password]
+  (let [system-time-µs (time/instant->micros (.getSystemTime tx-key))
+
+        live-table (.liveTable live-idx-tx user-table)
+        doc-writer (.docWriter live-table)]
+
+    (.logPut live-table (trie/->iid user) system-time-µs Long/MAX_VALUE
+             (fn write-doc! []
+               (.startStruct doc-writer)
+               (doto (.structKeyWriter doc-writer "_id" (FieldType/notNullable #xt.arrow/type :utf8))
+                 (.writeObject user))
+
+               (doto (.structKeyWriter doc-writer "username" (FieldType/notNullable #xt.arrow/type :utf8))
+                 (.writeObject user))
+
+               (doto (.structKeyWriter doc-writer "usesuper" (FieldType/notNullable #xt.arrow/type :bool))
+                 (.writeObject false))
+
+               (doto (.structKeyWriter doc-writer "passwd" (FieldType/nullable #xt.arrow/type :utf8))
+                 (.writeObject (util/md5 password)))
```

Comment:
let's go straight for bcrypt here otherwise it'll be a migration to do so

Reply:
- Should passwords be stored as bytes or as Strings (these seems to be some argument for bytes as Strings stick around longer on the JVM)
- What about the salt? Both bcrypt as well as argon2 need one.

Reply:
Should I also store the salt in `pg_user`?

Reply:
Resolved by using http://funcool.github.io/buddy-hashers/latest/user-guide.html